compare/src/compare-resources.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package src
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"k8s.io/kubernetes/test/e2e"
+
+	"github.com/golang/glog"
+)
+
+const (
+	// ResourceUsageVarianceAllowedPercent specifies how much variance we allow between results.
+	// Precise semantics is that we say that difference is too big if greater value is more than
+	// ResourceUsageVarianceAllowedPercent bigger than the smaller one. I.e. if X < Y, then Y is too
+	// big if Y > X + X*(ResourceUsageVarianceAllowedPercent/100).
+	ResourceUsageVarianceAllowedPercent = 50
+	// To avoid false negatives we assume that minimal CPU usage is 5% and memory 50MB
+	minCPU = 0.05
+	minMem = int64(50 * 1024 * 1024)
+)
+
+type percentileUsageData struct {
+	perc    string
```

Comment:
Don't get me started on 'why string'... It was an int, as it should be, but JSON encoder can't handle int keys in map. Hence it's just a serialized int, so I won't need it to cast it.

And it's abbreviated because I'm lazy - I can change that.